### ISSUE TITLE
Caja crashes while removing directory without permissions

### ISSUE THREAD

**Issue by zaps166** [mate-desktop/caja#483]
Sorry, but I have stripped binary, so I can't paste backtrace:
```
*** Error in `caja': double free or corruption (fasttop): 0x00007ff8f803cb10 ***
```
Steps to reproduce:

In terminal:
- mkdir testdir,
- chmod 000 testdir.

In Caja:
- select "testdir",
- press delete key.

Results: crash
Expected results: Caja shows that it can't remove directory or it forces to remove it.

OS: Manjaro/Arch Linux, Caja 1.12.0

**Comment by monsta** [comment]
Regression in GLib 2.46.1. Fixed in [this commit](https://git.gnome.org/browse/glib/commit/?h=glib-2-46&id=17177b81bf8b5be7121457c399e4b13cef0cfe34) which is included in 2.46.2.

**Comment by zaps166** [comment]
Ok, thanks!
